targets/TARGET_STM/TARGET_STM32L4/device/stm32l4xx_hal_spi.c

```diff
@@ -822,8 +822,12 @@ __IO uint16_t tmpreg;
       /* Check TXE flag */
       if((hspi->TxXferCount > 0) && ((hspi->Instance->SR & SPI_FLAG_TXE) == SPI_FLAG_TXE))
       {
-        hspi->Instance->DR = *((uint16_t *)hspi->pTxBuffPtr);
-        hspi->pTxBuffPtr += sizeof(uint16_t);
+        if (hspi->pTxBuffPtr) {
+          hspi->Instance->DR = *((uint16_t *)hspi->pTxBuffPtr);
+          hspi->pTxBuffPtr += sizeof(uint16_t);
+        } else {
+          hspi->Instance->DR = (uint16_t)0;
```

Comment:
Thats true, but if you want to analyze if the peripheral is really sending them (slave receives them), then its better to send something you can analyze rather than 0?